### PR TITLE
[#40] CI update, builds with IBM Semeru and Java 11, 17

### PR DESCRIPTION
**Pull request by bmarwell**
Some hints:

* Because these are a lot of builds, I did a pre-step with only Java 8 Temurin. If that fails, no need to build the others.
* The others use the matrix as expected. Excluded: The build which was already done.
* Build with IBM Semeru is a bit special. In contrast to Zulu and Temurin, it is not based on hotspot but Eclipse OpenJ9 and might have different encryption settings: algorithm aliases, defaults, etc.
Therefore I consider it highly valuable for anything security-related.
* Build with recent JDKs should be a no-brainer.
* I recommend to exclude some Mac builds to speed up the build.
* Semeru is not ready yet (https://github.com/actions/setup-java/pull/289), using adopt-openj9 until then.

### CODE REVIEW

**Comment by bmarwell** [comment]
The failing checks in 11/17:

```
 80 [INFO] --- keytool-maven-plugin:1.6-SNAPSHOT:generateKeyPair (default) @ MKEYTOOL-10 ---
 81 [INFO] /bin/sh -c cd '$HOME/git/mojohaus/keytool-maven-plugin/keytool-maven-plugin/target/it/MKEYTOOL-10' && '$HOME/.sdkman/candidates/java/17.0.2-tem/bin/keytool' '-genkeypair' '-v' '-keystore' '$HOME/git/mojohaus/keytool-maven-plugin/keytool-maven-plugin/target/it/MKEYTOOL-10/ta    rget/keystore.jks' '-storepass' 'secret' '-alias' 'alias' '-dname' 'cn=Test Development, ou=Development, L=Herzele, ST=VL, o=FrankCornelis, c=BE' '-keypass' 'secret'
 82 [INFO] Keytool-Fehler: java.lang.Exception: The -keyalg option must be specified.                                                                                                                                                                                                                                   
 83 [INFO] java.lang.Exception: The -keyalg option must be specified.
 84 [INFO]  at java.base/sun.security.tools.keytool.Main.doCommands(Main.java:1168)
 85 [INFO]  at java.base/sun.security.tools.keytool.Main.run(Main.java:415)
 86 [INFO]  at java.base/sun.security.tools.keytool.Main.main(Main.java:408)
```

Fix: `-keyalg` parameter.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -21,13 +21,39 @@ on: [push, pull_request]
...
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 8
```
Comment (slachiewicz) [inline .github/workflows/maven.yml:34]: Use 11 here

**Review by slachiewicz — APPROVED** [review]

File: .github/workflows/maven.yml (hunk 1 above)
Comment (bmarwell) [reply .github/workflows/maven.yml:34]: Can do. Any particular reason though?

Comment (slachiewicz) [reply .github/workflows/maven.yml:34]: Decrease usage of Java 8 😜 - 11 should be just faster. Personally i would use 17

**Comment by bmarwell** [comment]
@slachiewicz what do you think about excluding Zulu on Mac? 
Mac jobs tend to fail due to timeouts frequently.

**Comment by slachiewicz** [comment]
maybe let's use defaults for MacOS ie. Temurin? 
https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#java

**Comment by slachiewicz** [comment]
maybe let's use defaults for MacOS ie ~Temurin? 
https://github.com/actions/virtual-environments/blob/main/images/macos/macos-11-Readme.md#java

**Comment by olamy** [comment]
why do you merge that when the checks are still failing?

**Comment by bmarwell** [comment]
Because this PR was not about solving the Java 17 incompatibility, just about setting up the CI properly. See above, we identified the reason which is out of the scope of this PR. 

Will create another issue for the. 
But.
See the other issues: we are thinking about using the Keystore API instead of the executable. So even less reason to fix this one test here.

**Comment by olamy** [comment]
still doesn't make sense to merge a PR which breaks.
just fix the issues then merge it but the `master` in a working status.

**Comment by bmarwell** [comment]
It didn't break it. It wasn't working with JDK 17 before my PR. It just didn't show up.

**Comment by olamy** [comment]
I mean why adding 17 if it break??(https://github.com/mojohaus/keytool/commit/9e0adf3d0d28cd4cefd0a6c8dd33febf5067f8bc) 
if you make a requirement to build with 17 by adding it in the build file you must fix it.
if it's not in the build file it means it's not requirement (yet) but we know it doesn't work.
Now with the current situation if someone make a PR the PR will fail.

**Comment by bmarwell** [comment]
Oh I see what you mean.

I meant to fix it afterwards right away but had to go spontaneously. Will fix it later today. 😉

**Pull request merged**